frontend/packages/knative-plugin/src/components/add/event-sources/YAMLEditorSection.tsx

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { safeDump } from 'js-yaml';
+import { useFormikContext, FormikValues, useField } from 'formik';
+import YAMLEditorField from '@console/shared/src/components/formik-fields/YAMLEditorField';
+import FormSection from '@console/dev-console/src/components/import/section/FormSection';
+// eslint-disable-next-line import/no-cycle
+import { getEventSourceYaml } from '../../../utils/create-eventsources-utils';
+
+interface YAMLEditorSectionProps {
+  eventSourceType: string;
+}
+
+const YAMLEditorSection: React.FC<YAMLEditorSectionProps> = ({ eventSourceType }) => {
+  const { setFieldValue, setFieldTouched } = useFormikContext<FormikValues>();
+  const [namespaceField] = useField('project.name');
+  React.useEffect(() => {
+    setFieldValue('yamlData', safeDump(getEventSourceYaml(namespaceField.value, eventSourceType)));
+    setFieldTouched('yamlData', true);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why have you disabled the linting rule here?

Reply:
while using `useField`, the dependency was not allowing the component to load.
removed disable linting rule after updating code to take namespace as a prop